horace_core/sqw/PixelData/@MultipixBase/MultipixBase.m

```diff
@@ -0,0 +1,289 @@
+        end
+        %
+        %------------------------------------------------------------------
+        function npix = get.num_pixels(obj)
+            % total number of pixels in all contributing files
+            npix = obj.num_pixels_;
+        end
+        function npix_tot = get.npix_each_file(obj)
+            npix_tot = obj.npix_each_file_;
+        end
+        function obj= set.npix_each_file(obj,val)
+            % accepts the numeric array which defines number of pixels
+            % in each file or signle value if total number of pixels
+            % in each file is the same
+            obj = set_npix_each_file_(obj,val);
+        end
+        %------------------------------------------------------------------
+        function nb = get.nbins(obj)
+            nb = obj.nbins_;
+        end
+        function obj = set.nbins(obj,val)
+            % set number of bins property
+            % val -- single positive value, describing number of bins in
+            %        the images to be combined. Single value as have to be
+            %        same for all images
+            obj = set_nbins_(obj,val);
+        end
+        %
+        function range = get.pix_range(obj)
+            range = obj.data_range_(:,1:4);
+        end
+        function obj = set.pix_range(obj,val)
+            if ~(isnumeric(val) && isequal(size(val),[2,4]) )
+                error('HORACE:pix_combine_info:invalid_argument',...
+                    'pix_range size has to be a numeric array of 2x4 elements. It is:\n %s', ...
+                    disp2str(val));
+            end
+            obj.data_range_(:,1:4) = val;
+
+        end
+        %
+        function range = get.data_range(obj)
+            range = obj.data_range_;
+        end
+        function obj = set.data_range(obj,val)
+            if ~(isnumeric(val) && isequal(size(val),[2,9]) )
+                error('HORACE:pix_combine_info:invalid_argument',...
+                    'data_range size has to be numeric array of 2x9 elements. It is: %s', ...
+                    disp2str(val));
+            end
+            obj.data_range_ = val;
+        end
+        %
+        function md = get.metadata(obj)
+            md = pix_metadata(obj);
+        end
+        function dw = get.data_wrap(obj)
+            dw = pix_data(obj);
+        end
+        %
+        function fn = get.full_filename(obj)
+            fn = obj.full_filename_;
+        end
+        function obj = set.full_filename(obj,val)
+            if ~(ischar(val)||isstring(val))
+                error('HORACE:MultipixBase:invalid_argument', ...
+                    'fill_filename should be a string, describing full name of the file on disk. It is %s', ...
+                    disp2str(val));
+            end
+            obj.full_filename_ = val;
+        end
+        %
+        function is = get.is_filebacked(~)
+            is = true;
+        end
+        function is = get.is_misaligned(~)
+            is = false;
+        end
```

Comment:
Why are these `dependent`s and not `constant`s? Do we expect these to be overridden which I'm not sure is even legal in MATLAB? If this is `abstract` why are these not not implemented?

Reply:
because they are here to satisfy `PixelDatBase` interface